pkgs/applications/networking/instant-messengers/fractal-next/default.nix

```diff
@@ -0,0 +1,78 @@
+{ stdenv
+, lib
+, fetchFromGitLab
+, meson
+, ninja
+, rustPlatform
+, pkg-config
+, cmake
+, glib
+, gtk4
+, gtksourceview5
+, libadwaita
+, gstreamer
+, gst-plugins-base
+, gst-plugins-bad
+, libsecret
+, desktop-file-utils
+, appstream-glib
+, openssl
+, pipewire
+, llvmPackages
+, libshumate
+, wrapGAppsHook4
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "fractal-next";
+  version = "unstable-2022-07-10";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "GNOME";
+    repo = "fractal";
+    rev = "837b56978474fe512469805844b8ee234587499a";
+    hash = "sha256-6op/+eiDra5EFRludpkQOucBXdPl5a/oQWPwwhJEx+M=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) src;
+    hash = "sha256-2mE26ES+fYSWdfMr8uTsX2VVGTNMDQ9MXEk5E/L95UI=";
+  };
+
+  nativeBuildInputs = [
+    glib
+    gtk4
```

Comment:
Do glib and gtk4 need to be here?

Reply:
I assume they are for glib-compile-schemas and gtk4-update-icon-cache, which are used in [post install script](https://gitlab.gnome.org/GNOME/fractal/-/blob/837b56978474fe512469805844b8ee234587499a/meson.build#L90) :thinking:

Reply:
Yep, they’re unnecessary, I removed them.

Reply:
They should be there as Bobby points out. It only works because we do not have strictDeps for now, so they are accidentally picked from `buildInputs`.

Reply:
I added them back

Reply:
Just tested (with `strictDeps = true`), GTK isn't required to be in nativeBuildInputs.

Reply:
Removed it

Reply:
Actually, I tested too—and it breaks the alt-tab icon.

Reply:
Was that with `strictDeps = true`?

Reply:
Oh. No

Reply:
That's weird, you'd think it would still work since GTK would leak into PATH...

Reply:
Well, `strictDeps = true` makes it display a broken icon, while a missing `gtk4` just makes it show the default icon, and otherwise it shows a low-resolution icon.

Reply:
Interesting. Is there any combination of those things that makes it display the proper (presumably high-resolution) icon, or no?

Reply:
I assume the high-resolution icon is only displayed when the app is installed onto the system, as is my experience with most other packaged GNOME apps